### PR TITLE
Make links in description body clickable

### LINKED ISSUE #76: Edit Event Improvements
- [x] Allow storing HTML so that people can add links and images. From a PHP perspective, change `sanitize_text_field( $_POST['event_description'] )` to `make_clickable( $_POST['event_description'] )` so that URLs get autolinked. `wp_insert_post` will ensure that the `post_content` is sanitized.

### PR DESCRIPTION
**Pull request by trymebytes**
Fixes #76

### CODE REVIEW

**Comment by amieiro** [comment]
I am testing with 3 items:
1. A link.
2. An `<a>` item.
3. An `` item.

image

Only works the first item.

image

I don't know if it works correctly or not.

**Comment by trymebytes** [comment]
Nice catch, thank you!

File: templates/event.php [hunk 1]
```diff
@@ -34,7 +34,10 @@
 	</div>
 	<div class="event-details-left">
 		<div class="event-page-content">
-			<?php echo esc_html( $event_description ); ?>
+			<?php
+			// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+			echo make_clickable( wp_kses_post( $event_description ) );
```
Comment (akirk) [inline templates/event.php]: Why not this which would not require the override?
```suggestion
			echo wp_kses_post( make_clickable( $event_description ) );
```

File: wporg-gp-translation-events.php [hunk 2]
```diff
@@ -163,8 +163,10 @@ function submit_event_ajax() {
 		wp_send_json_error( 'Nonce verification failed' );
 	}
 
-	$title          = isset( $_POST['event_title'] ) ? sanitize_text_field( wp_unslash( $_POST['event_title'] ) ) : '';
-	$description    = isset( $_POST['event_description'] ) ? sanitize_text_field( wp_unslash( $_POST['event_description'] ) ) : '';
+	$title = isset( $_POST['event_title'] ) ? sanitize_text_field( wp_unslash( $_POST['event_title'] ) ) : '';
+	//@codingStandardsIgnoreStart
```
Comment (akirk) [inline wporg-gp-translation-events.php]: Is it an option to call `make_clickable()` already here? And why this style of disabling coding standards vs a `phpcs:ignore` style comment?

**Review by akirk — CHANGES_REQUESTED** [review]
I am not sold on the output escaping, maybe we can improve it

File: wporg-gp-translation-events.php (hunk 2 above)
Comment (trymebytes) [reply wporg-gp-translation-events.php]: After some tests, we could decide to call `make_clickable()` here or in the frontend it gives the same result. Do you suggest we rather call it here?

File: templates/event.php [hunk 3]
```diff
@@ -34,7 +34,9 @@
 	</div>
 	<div class="event-details-left">
 		<div class="event-page-content">
-			<?php echo esc_html( $event_description ); ?>
+			<?php
+				echo wp_kses_post( nl2br( make_clickable( $event_description ) ) );
```
Comment (akirk) [inline templates/event.php]: Does this also work?
```suggestion
				echo wp_kses_post( wpautop( make_clickable( $event_description ) ) );
```

Comment (akirk) [reply templates/event.php]: I tested and I think we should use this.

**Review by akirk — APPROVED** [review]
Ok for me now!

**Pull request merged**